src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,152 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { getSearchWith } from '../utils/searchHelper';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchParams(getSearchWith(searchParams,
+      { query: event.target.value || null }));
+  };
+
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries') || [];
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const loadPeople = async () => {
+    try {
+      setIsLoading(true);
+      const loadedPeople = await getPeople();
+
+      const peopleWithCenturies = loadedPeople.map(person => {
+        return { ...person, century: String(Math.ceil(person.died / 100)) };
+      });
+
+      setPeople(peopleWithCenturies);
+    } catch (error) {
+      setHasError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const visiblePeople = useMemo(() => {
+    let filteredPeople = people;
+
+    if (sex) {
```

Comment:
maybe lets do switch/case for sex, query and centuries?

Reply:
those are all separate filters that co-work i dont know how i can implement it with switch case.